### PR TITLE
Sqlite stream notify

### PR DESCRIPTION
**Pull request by skrytt**
for review only - few minor changes needed in addition to what is here

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -5,11 +5,13 @@ authors = ["skrytt"]
...
 
 [dependencies]
-env_logger = "0.6"
 dotenv = "0.14"
+env_logger = "0.6"
 log = "0.4"
 rand = "0.7"
-serenity = "0.6"
 reqwest = "0.9"
+rusqlite = "0.19"
```
Comment (Conundris) [inline Cargo.toml]: Is there a reason we not using the latest cargo release? (0.20)

File: src/main.rs [hunk 2]
```diff
@@ -38,15 +41,117 @@ impl EventHandler for Handler {
         info!("{} is connected!", ready.user.name);
     }
 
-    fn guild_member_update(&self,
-                           _context: Context,
-                           old_if_available: Option<Member>,
-                           new: Member)
+    fn presence_update(&self,
```
Comment (Conundris) [inline src/main.rs]: Main is starting to get big, possible to split event handling?

File: src/database.rs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        }
+    }
+    pub fn user_update(&self,
+        user_id: u64,
+        data: &User,
+    ) -> Result<(), ()>
+    {
+        let mut stmt = match self.connection.execute(
+            // Placeholder for storing something of actual use...
+            "INSERT OR REPLACE INTO Users(DiscordUserId) VALUES(?1)",
```
Comment (Conundris) [inline src/database.rs]: Not Applicable now but if you are to do commands which end up in as parameters need to be careful they checked etc.

File: src/database.rs [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            params![user_id as i64],
```
Comment (Conundris) [inline src/database.rs]: Maybe later on model structs for these for more general use? Not needed for presence_updates

File: src/database.rs [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+pub struct Handle {
+    connection: Connection,
+}
+
+impl Handle {
+    pub fn new(db_path: &str) -> Self {
+        Self {
+            connection: Connection::open(db_path)
```
Comment (Conundris) [inline src/database.rs]: https://sqlite.org/threadsafe.html
Something to be careful for is when we are having multiple connections to the db open. Might bring a potential danger with deadlocks and maybe overwriting (not 100% with sqlite). Need to be careful with more features using db.

File: Cargo.toml (hunk 1 above)
Comment (skrytt) [reply Cargo.toml]: Oops, wasn't intentional. I'll upgrade and test with the latest

**Pull request merged**